frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import { type PropsWithChildren, useEffect, useState } from "react";
+
+import { setLocaleHeader } from "metabase/lib/api";
+import { loadLocalization } from "metabase/lib/i18n";
+
+export const LocaleProvider = ({
+  children,
+  locale = "en",
+}: PropsWithChildren<{ locale?: string | null }>) => {
+  const [_isLoadingLocale, setIsLoadingLocale] = useState(false);
```

Comment:
I understand that we won't implement the loading state for the time being, so perhaps we can remove this unused loading state first

Reply:
It's actually useful as it forces a re-render, without this text could remain rendered with the wrong locale.
I thought there was a comment about it but i'll add it